changelogs/fragments/role_vars_scope_fix.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+    - Fix bug in `vars` applied to roles, they were being in correctly exported among others while only vars/main.yml was meant to be. Also adjusted the precedence to act the same as inline params.
```

Comment:
s/in correctly/incorrectly

It doesn't look like the precedence has been adjusted until the commented code is deleted:
```python
            # BCS NOTE: can remove here, were included above, this does change precedence
            if task._role:
                all_vars = _combine_and_track(all_vars, task._role.get_role_params(task.get_dep_chain()), "role '%s' params" % task._role.name)
```